src/lib.rs

```diff
@@ -808,32 +796,32 @@ impl Device {
 
     /// Creates a new buffer and maps it into host-visible memory.
     ///
-    /// This returns a [`CreateBufferMapped<T>`], which exposes a `&mut [T]`. The actual [`Buffer`]
+    /// This returns a [`CreateBufferMapped`], which exposes a `&mut [u8]`. The actual [`Buffer`]
     /// will not be created until calling [`CreateBufferMapped::finish`].
-    pub fn create_buffer_mapped<'a, T>(
-        &'a self,
-        count: usize,
-        usage: BufferUsage,
-    ) -> CreateBufferMapped<'a, T>
-    where
-        T: 'static + Copy + AsBytes + FromBytes,
-    {
-        let type_size = std::mem::size_of::<T>() as BufferAddress;
-        assert_ne!(type_size, 0);
+    pub fn create_buffer_mapped(&self, size: usize, usage: BufferUsage) -> CreateBufferMapped<'_> {
+        assert_ne!(size, 0);
 
         let desc = BufferDescriptor {
-            size: (type_size * count as BufferAddress).max(1),
+            size: size as BufferAddress,
             usage,
         };
         let mut ptr: *mut u8 = std::ptr::null_mut();
 
         let id = wgn::wgpu_device_create_buffer_mapped(self.id, &desc, &mut ptr as *mut *mut u8);
 
-        let data = unsafe { std::slice::from_raw_parts_mut(ptr as *mut T, count) };
+        let data = unsafe { std::slice::from_raw_parts_mut(ptr as *mut u8, size) };
 
         CreateBufferMapped { id, data }
     }
 
+    /// Creates a new buffer, maps it into host-visible memory, copies data from the given slice,
+    /// and finally unmaps it, returning a [`Buffer`].
+    pub fn create_buffer_mapped_from(&self, data: &[u8], usage: BufferUsage) -> Buffer {
```

Comment:
I think we should emove `mapped` from the name here, since the returned buffer is not in the mapped state. Perhaps, `create_buffer_with_data`?

Reply:
Done.